### PR TITLE
Add Immutable option to LTR Policy for SQL DB

### PR DESCRIPTION
**Pull request by rebeccaxu-ms**
### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [x] new API version
      - [x] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  4. By default, Azure SDKs of all languages (.NET/Python/Java/JavaScript for both management-plane SDK and data-plane SDK, Go for management-plane SDK only ) MUST be refreshed with/after swagger of new version is published. If you prefer NOT to refresh any specific SDK language upon swagger updates in the current PR, please leave details with justification here.

### Contribution checklist (MS Employees Only):
- [x] I commit to follow the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist

> **Applicability**: :warning:
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs
> - Adding new properties
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [x] Check this box if any of the following apply to the PR so that the label "ARMReview" and "WaitForARMFeedback" will be added by bot to kick off ARM API Review.  Missing to check this box in the following scenario may result in delays to the ARM manifest review and deployment.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[x] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits. You can use OpenAPIHub to initialize the PR for adding a new version. For more details refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version). Note that this doesn't apply if you are trying to merge a PR that was previously in the private repository.

- [x] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [x] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them.

### Breaking Change Review Checklist
If you have any breaking changes as defined in the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy/), request approval from the Breaking Change Review Board.

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Additional details on the process and office hours are on the [Breaking Change Wiki](https://eng.ms/docs/cloud-ai-platform/azure-core/azure-core-pm-and-design/trusted-platform-pm-karimb/service-lifecycle-and-actions-team/service-lifecycle-actions-team/apex/media/overview_breakingchanges).

NOTE: To update API(s) in public preview for over 1 year (refer to [Retirement of Previews](http://aka.ms/aprwiki))

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by v-xuto** [comment]
@JeffreyRichter Please help review the `breaking change` in this PR.

**Comment by ms-zhenhua** [comment]
@rebeccaxu-ms Please reach out to Azure Breaking Changes Reviewers azbreakchangereview@microsoft.com to get a breaking changes approval and ARM can review the changes afterwards. Once you have the approval, please remove the `ARMChangesRequested` label from the PR to make the PR visible to the ARM reviewers.

**Comment by JeffreyRichter** [comment]
You have to justify the break and adhere to Azure's Breaking Change policy.

**Review by ericshape — APPROVED** [review]

File: specification/sql/resource-manager/Microsoft.Sql/preview/2023-02-01-preview/LongTermRetentionPolicies.json
Comment (ms-zhenhua) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2023-02-01-preview/LongTermRetentionPolicies.json:232]: >weeklyRetention [](http://example.com/codeflow?start=9&length=15)

consider define "format": "date-time"

File: specification/sql/resource-manager/Microsoft.Sql/preview/2023-02-01-preview/LongTermRetentionPolicies.json
Comment (ms-zhenhua) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2023-02-01-preview/LongTermRetentionPolicies.json:236]: >monthlyRetention [](http://example.com/codeflow?start=9&length=16)

consider defining "format": "date-time"

File: specification/sql/resource-manager/Microsoft.Sql/preview/2023-02-01-preview/LongTermRetentionPolicies.json
Comment (ms-zhenhua) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2023-02-01-preview/LongTermRetentionPolicies.json:240]: >yearlyRetention [](http://example.com/codeflow?start=9&length=15)

consider defining "format": "date-time"

File: specification/sql/resource-manager/Microsoft.Sql/preview/2023-02-01-preview/LongTermRetentionPolicies.json
Comment (ms-zhenhua) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2023-02-01-preview/LongTermRetentionPolicies.json:246]: >in an ISO 8601 format [](http://example.com/codeflow?start=69&length=21)

format is int32 ?

File: specification/sql/resource-manager/Microsoft.Sql/preview/2023-02-01-preview/LongTermRetentionPolicies.json
Comment (ms-zhenhua) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2023-02-01-preview/LongTermRetentionPolicies.json:230]: >boolean [](http://example.com/codeflow?start=19&length=7)

consider using enum?
[https://github.com/Azure/azure-openapi-validator/blob/main/docs/enum-instead-of-boolean.md](https://github.com/Azure/azure-openapi-validator/blob/main/docs/enum-instead-of-boolean.md)

File: specification/sql/resource-manager/Microsoft.Sql/preview/2023-02-01-preview/LongTermRetentionPolicies.json
Comment (rebeccaxu-ms) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2023-02-01-preview/LongTermRetentionPolicies.json:232]: WeeklyRetention is in the form of PxM where x is the number of weeks.

File: specification/sql/resource-manager/Microsoft.Sql/preview/2023-02-01-preview/LongTermRetentionPolicies.json
Comment (rebeccaxu-ms) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2023-02-01-preview/LongTermRetentionPolicies.json:236]: monthlyRetention is in the form of PxY where x is the number of months

File: specification/sql/resource-manager/Microsoft.Sql/preview/2023-02-01-preview/LongTermRetentionPolicies.json
Comment (rebeccaxu-ms) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2023-02-01-preview/LongTermRetentionPolicies.json:240]: yearlyRetention is in the form of PxY where x is the number of years

File: specification/sql/resource-manager/Microsoft.Sql/preview/2023-02-01-preview/LongTermRetentionPolicies.json
Comment (rebeccaxu-ms) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2023-02-01-preview/LongTermRetentionPolicies.json:230]: There are only two options for makeBackupsImmutable, either yes or no

File: specification/sql/resource-manager/Microsoft.Sql/preview/2023-02-01-preview/LongTermRetentionPolicies.json
Comment (rebeccaxu-ms) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2023-02-01-preview/LongTermRetentionPolicies.json:246]: This is not actually in an ISO 8601 format, I can look into fixing this in a later PR but I would like to unblock this PR before the deadline on Friday

**Comment by ms-zhenhua** [comment]
@rebeccaxu-ms signed off. But please remember to update the descriptions for the **Retention** properties

**Comment by v-xuto** [comment]
@raych1 Please help review Go breaking change.

**Comment by v-xuto** [comment]
@raych1 Please help review Go breaking change.

**Pull request merged**
